recipe/bld.bat

```diff
@@ -24,22 +24,24 @@ cmake -G "Ninja" ^
       -DARROW_JSON=ON ^
       -DARROW_MIMALLOC=OFF ^
       -DARROW_OPENSSL_USE_SHARED=ON ^
-      -DARROW_ORC=ON ^
+      -DARROW_ORC=OFF ^
       -DARROW_PACKAGE_PREFIX="%LIBRARY_PREFIX%" ^
       -DARROW_PARQUET=ON ^
       -DARROW_PLASMA=ON ^
       -DARROW_PROTOBUF_USE_SHARED=ON ^
-      -DARROW_PYTHON=ON ^
       -DARROW_SIMD_LEVEL=DEFAULT ^
       -DARROW_S3=ON ^
       -DARROW_WITH_BROTLI=ON ^
       -DARROW_WITH_BZ2=ON ^
       -DARROW_WITH_LZ4=ON ^
-      -DARROW_WITH_SNAPPY=ON ^
```

Comment:
Why have we turned off support for ORC, Python, and Snappy? We should not turn off features that were in previous releases without warning end users. (Plus Python support is going to be needed elsewhere.)

Reply:
In this case `ORC` support is turned off only for the windows builds because it is causing issues with `pthread` in the build, the solution for that problem was to turn off `ORC`

In addition conda-forge also had to set `ORC` off in order to build for windows.

https://github.com/conda-forge/arrow-cpp-feedstock/blob/1d6fb32594ab40dec054e0b4844861dcaa5c042d/recipe/bld-arrow.bat#L34

`ARROW_Python` is deprecated in version 10.0.0 of `arrow_cpp` The windows builds fail in the new versions when `ARROW_Python` is said to on.

The upstream documentation explains that point in a bit more detail:

https://arrow.apache.org/docs/developers/cpp/building.html#:~:text=%2DDARROW_PYTHON%3DON%3A%20This%20option%20is%20deprecated%20since%2010.0.0.%20This%20will%20be%20removed%20in%20a%20future%20release.%20Use%20CMake%20presets%20instead.%20Or%20you%20can%20enable

Also adding `Snappy` to the recipe in windows will generate an overlinking error. When `Snappy` was set to off the package was able to build.

Reply:
Instead of `-DARROW_PYTHON=ON` you can enable ARROW_COMPUTE, ARROW_CSV, ARROW_DATASET, ARROW_FILESYSTEM, ARROW_HDFS, and ARROW_JSON directly instead.
But  `-DARROW_COMPUTE=ON` and `-DARROW_FILESYSTEM=ON` are missing.

Reply:
`-DARROW_COMPUTE=ON` is in the `bld.bat` file

https://github.com/AnacondaRecipes/arrow-cpp-feedstock/blob/3174d297567e835d0a260d556286039d6dfb384a/recipe/bld.bat#L16

`-DARROW_FILESYSTEM=ON` is in the `bld.bat` file

https://github.com/AnacondaRecipes/arrow-cpp-feedstock/blob/3174d297567e835d0a260d556286039d6dfb384a/recipe/bld.bat#L21